transcom/mymove
# B 19800 int Zip/City lookup on Customer Profile

## [B-19800](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A946208)

## Summary
This backlog adds a Zip/City lookup capability to the Customer Profile address pages. Selecting a result populates the now read only City, State, County, and Zip fields. To make this possible a new endpoint was added that will query the us_post_region_cities database table and return results that have the search criteria in either the zip code or the city name. 

![image](https://github.com/user-attachments/assets/4ba8b076-c4fb-4d18-8d55-073757c74a19)

Is there anything you would like reviewers to give additional scrutiny?
More pages will be modified as downstream tasks for other Customer pages and Office user pages so a lot of this functionality will be re-used there. If you can think of any changes that may make the functionality more modular I am open to suggestions.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login to the Customer app and create a new user
2. Fill out the pages until you come to the Customer Address page
3. You should now see a Zip/City lookup field and the City, State, County, Zip fields are read only
![image](https://github.com/user-attachments/assets/05ebc2b0-ddf7-4aee-8002-f2257c30c43e)
4. Enter an address for Address 1
5. Search for a zip or city and select an result
6. The City, State, County, and Zip fields should all populate
7. Once you enter the address and select a Zip/City result the Next button should be visible to click
8. Click Next
9. Perform the similar actions as on the Customer Address page for the Backup Address page and verify you can click Next
10. Go back to the Backup and Customer Address pages to verify the data you selected earlier is still populated

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
